sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1749,6 +1749,19 @@ object SQLConf {
     .checkValue(v => v > 0, "The min partition number must be a positive integer.")
     .createOptional
 
+  val FILES_MAX_DESIRED_PARTITION_NUM = buildConf("spark.sql.files.maxDesiredPartitionNum")
+    .doc("The maximum desired number of partitions when reading files. When the number of " +
+      "partitions calculated for the first time is greater than this value, recalculate " +
+      s"${FILES_MAX_PARTITION_BYTES.key} so that the final number of partitions is close to this " +
+      "value. Note that the final calculated number of partitions may be larger than this value." +
+      "This configuration is effective only when using file-based sources such as Parquet, JSON " +
+      "and ORC.")
+    .version("3.5.0")
+    .intConf
+    .checkValue(threshold => threshold > 0,
```

Comment:
should we change to check `threshold > FILES_MIN_PARTITION_NUM`?

Reply:
According to the document of `FILES_MIN_PARTITION_NUM`, it seems `FILES_MIN_PARTITION_NUM` just suggests(not guaranteed) minimum number of split file partitions .
Maybe we can't rely on it.